cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -12,23 +18,41 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
 
   import configuration.{includeStack, includeThread}
 
-
+  var classLoaderKey = 0
 
   def parse(): Unit = {
 
     println(s"*** File $file")
     var allocationInNewTlab: Boolean = false
     var allocationOutsideTlab: Boolean = false
     var methodProfilingSample: Boolean = false
+    var classPaths =  mutable.ArraySeq[String]()
+
+    def extractClassPath(event: RecordedEvent): Unit = {
+      event.getFields.forEach(
+        x => {
+          if(event.getValue("key").equals("java.class.path")) {
+            val clazzpaths:String = event.getValue("value")
+            val delim = if (System.getProperty("os.name").toLowerCase.contains("win")) ";" else ":"
+            val urls = clazzpaths.split(delim)
+            classPaths = classPaths.:++(urls)
+          }
+        }
+      )
+    }
+
     try {
       val recordingFile = new RecordingFile(file.toPath)
       //      for (event <- view.asScala) {
+      classLoaderKey = file.getPath.hashCode;
```

Comment:
hashcode can collide.
We also want to share info based on the classpaths being the same

The "key" here is the "java.class.path" recorded in the JFR file system properties event

Reply:
the filename information is queried after all jfr files has been processed so having the key as "java.class.path" will collide with different jfr files. also I cannot find the "sun.boot.class.path".